File: frontend/lib/features/credit/domain/entities/customer.dart
// The Customer entity represents a person or business who shops at the store.
// It tracks their profile and any outstanding debt (credit).
class Customer {
  final String id;
  final String name;
  final String phone;
  final String imageUrl;
  final double totalOutstanding;
  final double creditLimit;
  final String status;
  final String lastPurchase;
  final String createdAt;

  Customer({
    required this.id,
    required this.name,
    this.phone = '',
    this.imageUrl = '',
    this.totalOutstanding = 0, // The current amount of money this customer owes the shop
    this.creditLimit = 5000, // The maximum debt this customer is allowed to incur
    this.status = 'active',
    this.lastPurchase = '',
    this.createdAt = '',
  });

  factory Customer.fromJson(Map<String, dynamic> json) {
    return Customer(
      id: json['id'] ?? '',
      name: json['name'] ?? '',
      phone: json['phone'] ?? '',
      imageUrl: json['imageUrl'] ?? '',
      totalOutstanding:
          double.tryParse(json['totalOutstanding']?.toString() ?? '0') ?? 0.0,
      creditLimit:
          double.tryParse(json['creditLimit']?.toString() ?? '0') ?? 0.0,
      status: json['status'] ?? 'active',
      lastPurchase: json['lastPurchase'] ?? '',
      createdAt: json['createdAt'] ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {'name': name, 'phone': phone, 'creditLimit': creditLimit};
  }
}

File: backend/src/domain/entities/Customer.js
// The Customer entity represents a person or business that buys from the shop, often on credit.
class Customer {
    constructor({ 
        id, 
        name, 
        phone, 
        imageUrl, 
        totalOutstanding, // The amount of money this customer currently owes the shop
        creditLimit, // The maximum amount of debt this customer is allowed to have
        status, 
        lastPurchase, 
        createdAt, 
        updatedAt 
    }) {
        this.id = id;
        this.name = name || '';
        this.phone = phone || '';
        this.imageUrl = imageUrl || '';
        this.totalOutstanding = totalOutstanding || 0;
        this.creditLimit = creditLimit || 0;
        this.status = status || 'active';
        this.lastPurchase = lastPurchase || '';
        this.createdAt = createdAt || new Date().toISOString();
        this.updatedAt = updatedAt || new Date().toISOString();
    }

    toJSON() {
        return {
            id: this.id,
            name: this.name,
            phone: this.phone,
            imageUrl: this.imageUrl,
            totalOutstanding: this.totalOutstanding,
            creditLimit: this.creditLimit,
            status: this.status,
            lastPurchase: this.lastPurchase,
            createdAt: this.createdAt,
            updatedAt: this.updatedAt,
        };
    }
}

module.exports = Customer;
